dist/bin/dotr

```diff
@@ -31,18 +31,39 @@ source "$PROG_HOME/bin/common"
 
 CLASS_PATH=".$PSEP$DOTTY_LIB$PSEP$SCALA_LIB"
 
-# -d must be the first option
-case "$1" in
-  -d) DEBUG="$DEBUG_STR" && shift ;;
-esac
 
-first_arg="$1"
+addResidual () {
+  residual_args+=("$1")
+}
 
-if [ -z "$1" ]; then
+run_repl=true
+run_app=flase
+while [[ $# -gt 0 ]]; do
+key=$1
+case $key in
+        -repl)
+         run_repl=true
+         shift
+         ;;
+        -classpath)
+         CLASS_PATH+=":$2"
```

Comment:
We should probably add the user-defined classpaths to the left of the classpath to give them the priority.